### PR TITLE
ci: ensure all tool maintainers are assignable on issues

### PR DESCRIPTION
**Pull request by emilyalbini**
GitHub only allows people explicitly listed as collaborators on the repository or who commented on the issue/PR to be assignees, failing to create the issue if non-assignable people are assigned.

This adds an extra check on CI to make sure all the people listed as tool maintainers can be assigned to toolstate issues. The check won't be executed on PR builds due to the lack of a valid token.

r? @kennytm

### CODE REVIEW

**Comment by pietroalbini** [comment]
We can't merge this yet as CI will fail due to these errors:

```
rust-by-example: @marioidival is not assignable
rust-by-example: @projektir is not assignable
embedded-book maintainer @adamgreig is not assignable
embedded-book maintainer @andre-richter is not assignable
embedded-book: @jamesmunns is not assignable
embedded-book: @korken89 is not assignable
embedded-book: @ryankurte is not assignable
embedded-book: @thejpster is not assignable
embedded-book: @therealprof is not assignable
rustc-guide: @mark-i-m is not assignable
rustc-guide: @amanjeev is not assignable
rustfmt: @topecongiro is not assignable
```

**Comment by pietroalbini** [comment]
For everyone pinged by this comment: toolstate is configured to open a new issue when a tool breaks, assigning the people who usually fix those breakages. Unfortunately GitHub only allows assigning people with explicit access to a repo, so we need to grant everyone who's going to be assigned access to the repo.

We're going to grant "read" access to the repo, which doesn't add new real permissions (well, it's a public repository...) while allowing y'all to be assigned. Since adding permissions to individuals is not a good practice we should get GitHub teams in the rust-lang org for each tool and then grant access to those teams.

---

Opened PRs in the team repo to synchronize [wg-rustfmt](https://github.com/rust-lang/team/pull/105) and [rust-by-example](https://github.com/rust-lang/team/pull/106) with GitHub.

@nikomatsakis which teams should @mark-i-m and @amanjeev be part of for rustc-guide?

@therealprof could you put the people responsible for the embedded book in a separate team/wg on the team repo? Then we can automatically synchronize that team with the rust-lang GitHub organization.

**Comment by therealprof** [comment]
@pietroalbini The @rust-embedded/resources team is responsible for managing the book, is that granular enough for this purpose?

**Comment by pietroalbini** [comment]
@therealprof yeah, but it needs to be on the `rust-lang` org. If you setup the team on the rust-lang/team repo we can synchronize it with GitHub.

File: src/tools/publish_toolstate.py
Comment (kennytm) [inline src/tools/publish_toolstate.py]: Maybe `MAINTAINERS` should be changed to an dictionary of list/set?

**Comment by therealprof** [comment]
@pietroalbini  Like https://github.com/rust-lang/team/pull/107 ?

**Comment by pietroalbini** [comment]
Yep! rustfmt and embedded-book should now be fixed.

File: src/tools/publish_toolstate.py
Comment (pietroalbini) [reply src/tools/publish_toolstate.py]: That's actually intentional, yeah, I don't want the rest of the script to run when validation completes.

**Comment by pietroalbini** [comment]
@kennytm addressed review comments.

**Review by kennytm — APPROVED** [review]

**Comment by kennytm** [comment]
r=me after everyone is confirmed assignable.

File: src/tools/publish_toolstate.py
Comment (RalfJung) [reply src/tools/publish_toolstate.py]: Might be worth adding a comment explaining this? Or moving the exit call out to where this function gets called?

Comment (pietroalbini) [reply src/tools/publish_toolstate.py]: [Already added that line along with the other fixes.](https://github.com/rust-lang/rust/pull/64119/files#diff-cc999bf51e313e1dc34a936abc30a10dR260)

**Comment by pietroalbini** [comment]
Also removed `@projektir` as they don't contribute to rust by example anymore. People who still need to get permissions:

* rust-by-example: `@marioidival` needs to accept the invitation to join the org
* rustc-guide: https://github.com/rust-lang/team/pull/112 needs to be merged and then the members needs to join the org

**Comment by pietroalbini** [comment]
All the permissions are properly configured :tada: 

@bors r=kennytm

**Pull request merged**
